src/query/users/src/role_mgr.rs

```diff
@@ -78,17 +79,26 @@ impl UserApiProvider {
     // Ensure the builtin roles inside a tenant. Currently we have two builtin roles:
     // 1. ACCOUNT_ADMIN, which has the equivalent privileges of `GRANT ALL ON *.* TO ROLE account_admin`,
     //    it also contains all roles. ACCOUNT_ADMIN can access the data objects which owned by any role.
-    // 2. PUBLIC, which have no any privilege by default, but every role contains the PUBLIC role.
-    //    The data objects which owned by PUBLIC can be accessed by any role.
+    // 2. PUBLIC, on the other side only includes the public accessible privileges, but every role
+    //    contains the PUBLIC role. The data objects which owned by PUBLIC can be accessed by any role.
     pub async fn ensure_builtin_roles(&self, tenant: &str) -> Result<u64> {
-        let mut role_info = RoleInfo::new(BUILTIN_ROLE_ACCOUNT_ADMIN);
-        role_info.grants.grant_privileges(
+        let mut account_admin = RoleInfo::new(BUILTIN_ROLE_ACCOUNT_ADMIN);
+        account_admin.grants.grant_privileges(
             &GrantObject::Global,
             UserPrivilegeSet::available_privileges_on_global(),
         );
-        self.add_role(tenant, role_info, true).await?;
-        let role_info = RoleInfo::new(BUILTIN_ROLE_PUBLIC);
-        self.add_role(tenant, role_info, true).await
+        self.add_role(tenant, account_admin, true).await?;
+
+        let mut public = RoleInfo::new(BUILTIN_ROLE_PUBLIC);
+        public.grants.grant_privileges(
+            &GrantObject::Table(
+                "default".to_string(),
```

Comment:
We can define a system database list and each item has what's a role that can grant it, it easy to maintain.

Reply:
currently we only have two roles builtin `account_admin` and `public`, these two roles are both special:

1. `account_admin` can access every tables
2. `public` is supposed to not able to access any system table at all.

We translate `SELECT 1` to query an internal table `system.one`, this makes us have to grant a privilege on `system.one` for every user. However the query `SELECT 1` do not really touches any tables in the query semantics. So I guess `system.one` can be viewed as a special corner case?